src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -2337,4 +2430,121 @@ RppStatus rppt_pixelate_gpu(RppPtr_t srcPtr,
 #endif // backend
 }
 
+/******************** fog ********************/
+
+RppStatus rppt_fog_gpu(RppPtr_t srcPtr,
+                       RpptDescPtr srcDescPtr,
+                       RppPtr_t dstPtr,
+                       RpptDescPtr dstDescPtr,
+                       Rpp32f *gammaTensor,
+                       RpptROIPtr roiTensorPtrSrc,
+                       RpptRoiType roiType,
+                       rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+
+    RpptInterpolationType interpolationType = RpptInterpolationType::NEAREST_NEIGHBOR;
+    RpptDesc fogMaskSrcDesc;
+    RpptDescPtr fogMaskSrcDescPtr = &fogMaskSrcDesc; // Fog original mask source description pointer
+    set_fog_mask_descriptor(fogMaskSrcDescPtr, 2, FOG_MAX_HEIGHT, FOG_MAX_WIDTH, 1);  
+
+    RpptDesc fogMaskDstDesc;
+    RpptDescPtr fogMaskDstDescPtr = &fogMaskDstDesc; //Fog resized mask destination description pointer
+    set_fog_mask_descriptor(fogMaskDstDescPtr, 2, srcDescPtr->h, srcDescPtr->w, 1);
+
+    RpptImagePatchPtr internalDstImgSizes = reinterpret_cast<RpptImagePatch *>(rpp::deref(rppHandle).GetInitHandle()->mem.mgpu.scratchBufferPinned.floatmem);
+    RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + 2);
+
+    for(Rpp32s i = 0; i < 2; i++)
+    {
+        internalDstImgSizes[i] = {srcDescPtr->w, srcDescPtr->h};
+        internalRoiTensorPtrSrc[i].xywhROI = {0, 0, 1920, 1080};
+    }
+    rppSetBatchSize(rppHandle,2);                   // Setting batch size to 2 for computing resizing for alpha and intensity
```

Comment:
Please add space after ,
```suggestion
    rppSetBatchSize(rppHandle, 2);                   // Setting batch size to 2 for computing resizing for alpha and intensity
```